assets/src/edit-story/components/panels/sizePosition.js

```diff
@@ -109,21 +109,78 @@ function SizePositionPanel({ selectedElements, onSetProperties }) {
           return update;
         }
       );
+    },
+    [canFlip, lockRatio, onSetProperties]
+  );
+
+  const handleSubmit = useCallback(
+    (evt) => {
+      updateProperties(state);
       if (evt) {
         evt.preventDefault();
         evt.stopPropagation();
       }
     },
-    [canFlip, lockRatio, onSetProperties, state]
+    [updateProperties, state]
   );
 
-  const handleNumberChange = useCallback(
-    (property) => (value) =>
-      setState((originalState) => ({
-        ...originalState,
-        [property]: isNaN(value) || value === '' ? '' : parseFloat(value),
-      })),
-    [setState]
+  const handleWidthChange = useCallback(
+    (value) => {
+      const ratio = width / height;
+      const newWidth = isNaN(value) || value === '' ? '' : parseFloat(value);
+
+      setState((originalState) => {
+        const update = {
+          ...originalState,
+          width: newWidth,
+          height:
+            height !== '' && typeof newWidth === 'number' && lockRatio
+              ? dataPixels(newWidth / ratio)
+              : height,
+        };
+        updateProperties(update);
+
+        return update;
+      });
```

Comment:
So, you're both setting state locally **and** updating state? But won't updating properties also [update state as the new elements come in](https://github.com/google/web-stories-wp/pull/526/files#diff-5aaebc56e19801cc9d7f2c637c34a026R74-R77)?

Reply:
I think you might be able to completely remove `state` from this component, as all changes are persisted upstream now.

Reply:
Interesting. Yeah, let me try something in this way. I think that is totally possible.

Reply:
It's possible to remove the state from this panel, but *only* if updating the timeout here https://github.com/google/web-stories-wp/pull/526/files#diff-c1b33953127f7a773e4406edd5c27bd6R38 from `300` to `0`. It's okay if the user types slowly, but if the user types fastly or even press and hold some key, it will have a delayed feedback experience on the input field. WDYT?

Reply:
Ah yes of course, then React will think the components are uncontrolled, makes sense. Yeah, you need to update local state immediately and upstream state can then debounce (and also update local state delayed, but that's insignificant).

Reply:
Yup. Exactly what is happening.

Reply:
We really should move the "debouncing" to the render in canvas and page preview (and even debounce even slower in page preview, as it's even less important to be "live") and always sync panel state to reducer state instantly. But that's a future improvement.